docs/_snippets/automocked-modules-in-story.md

```diff
@@ -0,0 +1,282 @@
+```ts filename="AuthButton.stories.ts" renderer="angular" language="ts"
+import type { Meta, StoryObj } from '@storybook/angular';
+import { expect, mocked } from 'storybook/test';
+
+import { AuthButton } from './AuthButton.component';
+
+import { v4 as uuidv4 } from 'uuid';
+import { getUserFromSession } from '../lib/session';
+
+const meta: Meta<AuthButton> = {
+  component: AuthButton,
+  // 👇 This will run before each story is rendered
+  beforeEach: async () => {
+    // 👇 Force known, consistent behavior for mocked modules
+    mocked(uuidv4).mockReturnValue('1234-5678-90ab-cdef');
+    mocked(getUserFromSession).mockReturnValue({ name: 'John Doe' });
+  },
+};
+export default meta;
+
+type Story = StoryObj<AuthButton>;
+
+export const LogIn: Story = {
+  play: async ({ canvas, userEvent }) => {
+    const button = canvas.getByRole('button', { name: 'Sign in' });
+    userEvent.click(button);
+
+    // Assert that the getUserFromSession function was called
+    expect(getUserFromSession).toHaveBeenCalled();
+  },
+};
+```
+
+```ts filename="AuthButton.stories.ts" renderer="common" language="ts"
+// Replace your-framework with the name of your framework (e.g. react-vite, vue3-vite, etc.)
+import type { Meta, StoryObj } from '@storybook/your-framework';
+import { expect, mocked } from 'storybook/test';
+
+import { AuthButton } from './AuthButton';
+
+import { v4 as uuidv4 } from 'uuid';
+import { getUserFromSession } from '../lib/session';
+
+const meta = {
+  component: AuthButton,
+  // 👇 This will run before each story is rendered
+  beforeEach: async () => {
+    // 👇 Force known, consistent behavior for mocked modules
+    mocked(uuidv4).mockReturnValue('1234-5678-90ab-cdef');
+    mocked(getUserFromSession).mockReturnValue({ name: 'John Doe' });
+  },
+} satisfies Meta<typeof AuthButton>;
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const LogIn: Story = {
+  play: async ({ canvas, userEvent }) => {
+    const button = canvas.getByRole('button', { name: 'Sign in' });
+    userEvent.click(button);
+
+    // Assert that the getUserFromSession function was called
+    expect(getUserFromSession).toHaveBeenCalled();
+  },
+};
+```
+
+```js filename="AuthButton.stories.js" renderer="common" language="js"
+import { expect } from 'storybook/test';
```

Comment:
@kylegach, the JS variants are not similar to the TS variants. There's a missing `mocked` import from `storybook/test`. Also, the `uuidv4` should be wrapped in the `mocked` function call.

Reply:
The `mocked` import and function call is only needed for types. So I kept the JS ones simplified.